### ISSUE TITLE
Differentiating sport infrastructure (like pitches) by type of sport

### ISSUE THREAD

**Issue by matthijsmelissen** [gravitystorm/openstreetmap-carto#844]
Currently all sport fields are shown in the same style. It is very difficult to distinguish the different sports. Furthermore there are good sport icon sets out there:  http://wiki.openstreetmap.org/wiki/Category:Sport_icon Comparing the different olympic pictograms  http://www.nytimes.com/interactive/2010/02/24/sports/olympics/pictograms-interactive.html I suggest to take the icon sets of Olympic games 1972 (munich)  http://olympic-museum.de/pictograms/Picto1972.htm adopted to the constraints of 16px 

See also https://trac.openstreetmap.org/ticket/3159.

**Comment by boothym** [comment]
How about adding icons for the most used sports to begin with? Soccer, tennis, baseball, basketball, all have good icons that can be used from iD editor's icon set. Just like iD the icon can be put in the middle of the pitch.

https://github.com/bhousel/maki/blob/mb-pages/src/soccer-24.svg
https://github.com/bhousel/maki/blob/mb-pages/src/tennis-24.svg
https://github.com/bhousel/maki/blob/mb-pages/src/baseball-24.svg
https://github.com/bhousel/maki/blob/mb-pages/src/basketball-24.svg

**Comment by kocio-pl** [comment]
Unfortunately they all are 24px, while we use just 14px - this is huge difference. If you want to see how they would look in this style, try to rescale them and save as PNG.

**Comment by boothym** [comment]
I linked to them because I thought you'd want the largest svg size available, but here's what they look like at 14px:
![tennis](https://cloud.githubusercontent.com/assets/16965772/18996247/7b0ce524-8727-11e6-88d5-172585b3e489.png)
![soccer](https://cloud.githubusercontent.com/assets/16965772/18996248/7b0f6196-8727-11e6-86f2-0e5233a26828.png)
![baseball](https://cloud.githubusercontent.com/assets/16965772/18996249/7b0f9026-8727-11e6-912c-14cb62564035.png)
![basketball](https://cloud.githubusercontent.com/assets/16965772/18996246/7b0cc076-8727-11e6-822c-88bc39b9384e.png)

There's also https://github.com/mapbox/maki/tree/master/icons with more stick figures instead of the bat/ball icons.

**Comment by kocio-pl** [comment]
I'm surprised, they look quite nice and are perfectly recognizable! That's the power of skilled designer probably. =}

Still we need to decide if we want to use icons or field markings.

**Comment by Wuzzy2** [comment]
I'm very strongly for icons. Field markings are poorly recognizable and potentionally ambigious. Also, the field markings are not actually based on OSM data so they are basically lies, stricly spoken. The real field markings might either not exist at all or they are just wrong.
For example, not all soccer fields actually have field markings, especially small ones. Also, the field markings will be hard to be understood for people not knowing a particular sport very well.

On the other hand, icons are WAY more recognizable, even by people not knowing awfully much about the sport.

**Comment by HolgerJeromin** [comment]
> For example, not all soccer fields actually have field markings, especially small ones. 

Maps are no rendered satellite images.

> Also, the field markings will be hard to be understood for people not knowing a particular sport very well.
> On the other hand, icons are WAY more recognizable, even by people not knowing awfully much about the sport. 

As an counter example there are [many racket sports](https://en.wikipedia.org/wiki/List_of_racket_sports). I am not sure which is meant by the tennis icon from boothym. But the line marking is pretty unique.

> Also, field markings do not fit into the “tradition” of the default OSM style of only showing what's actually there in the data, rather than “interpreting” stuff into the data which simply isn't there.

Think of the markings as a big icon :-P

**Comment by Wuzzy2** [comment]
> > For example, not all soccer fields actually have field markings, especially small ones.

> Maps are no rendered satellite images.

Exactly. And I think field markings try way too hard to look “authentic”. Under the same logic, OSM could attempt to draw road markings on roads.
Icons are more abstract, therefore more logical for a map.

> As an counter example there are many racket sports. I am not sure which is meant by the tennis icon from boothym. But the line marking is pretty unique.

I guess the tennis icon stands for “tennis”. ;-) You already said “tennis” by yourselves! xD Also because this is probably the most well-known one.
Yes, it's probably hard to draw an icon for every sport in existance but OSM could just start with the most common ones while adding icons later. Sports without an icon could maybe get a generic icon or no icon.

Maybe the homepage should improve its map key because the default style has so many different icons, but almost no icon is actually explained. :-(
Creating a key for the icons would (hopefully) not be too hard (it just needs to be done).
But creating a key for the markings or all sport pitches would be overkill and therefore it would be pretty unlikely to happen. I fear that Average Joe will fail to recognize the lion's share of possible sport pitches.

> Think of the markings as a big icon :-P

Pfff, this is weak. The markings are nothing like icons. Icons are not rotated and have a fixed size and are always put on a point. Markings can be rotated, are made of lines and differ in size and take an area.

**Comment by boothym** [comment]
> As an counter example there are many racket sports. I am not sure which is meant by the tennis icon from boothym. But the line marking is pretty unique.

Other racket sports have slightly different rackets (e.g. longer handle in badminton) and most of them are played indoors. If you see a pitch in OSM with a racket and ball, and it's not inside a building it's highly likely to be tennis (given 230,000 courts have been mapped!).

Anyway, the main thing is that I think icons should be added as soon as possible for the most important sports in OSM. They presumably are easier to add than markings and would improve the map styling.

**Comment by boothym** [comment]
Is someone able to render these so we can see what they look like? @kocio-pl ?

**Comment by kocio-pl** [comment]
What should the colors be (re)used then?

I hope pitch color will be made [less dominant](https://github.com/gravitystorm/openstreetmap-carto/pull/2363) soon, but green symbols with green label could be hardly readable.

**Comment by boothym** [comment]
I was thinking basically just how they are - dark gray/black (with transparency)?

**Comment by Hufkratzer** [comment]
Can you please also add an icon for **riding arenas** (tagged [leisure=pitch](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch) + [sport=equestrian](https://wiki.openstreetmap.org/wiki/Tag:sport%3Dequestrian) :horse:) ?

The [taginfo statistics for leisure=pitch + sport=*](http://taginfo.openstreetmap.org/tags/leisure=pitch#combinations) is:
soccer; 276k
tennis: 234k
baseball: 91k
basketball: 76k
multi: 32k
_golf: 23k (but golf courses shall not be tagged as pitches)_
**equestrian: 15k** :horse:
volleyball: 12k
athletics: 7k

The icon may be:
a) horse shoe
b) horse head :horse: _<- github horse head_
c) horse without rider
d) horse with rider :horse_racing: _<- github horse rider_

For a) and b) here are two possibilities shown together with the other icons:
![hshoe-14](https://cloud.githubusercontent.com/assets/5958184/22522126/c872539a-e8ba-11e6-912d-86bd2076643b.png) from http://publicdomainvectors.org/en/free-clipart/Horseshoe-silhouette-vector-graphics/14822.html
![hhead14-1](https://cloud.githubusercontent.com/assets/5958184/22512328/eed19da6-e898-11e6-80b7-78c50006b500.png) from https://commons.wikimedia.org/wiki/File:1_Cav_Shoulder_Insignia.svg
![mysoccer-14](https://cloud.githubusercontent.com/assets/5958184/22522184/04b953bc-e8bb-11e6-8b2b-02be44077fb5.png) from https://github.com/bhousel/maki/blob/mb-pages/src/soccer-24.svg
![mytennis-14](https://cloud.githubusercontent.com/assets/5958184/22512329/ef38abf4-e898-11e6-8d91-15c941174856.png) from https://github.com/bhousel/maki/blob/mb-pages/src/tennis-24.svg
![mybaseball-14](https://cloud.githubusercontent.com/assets/5958184/22522286/544e6ebc-e8bb-11e6-902a-e786529f248b.png) from https://github.com/bhousel/maki/blob/mb-pages/src/baseball-24.svg
![mybasketball-14](https://cloud.githubusercontent.com/assets/5958184/22522323/72da5e4a-e8bb-11e6-9460-bc8023f889e9.png) from https://github.com/bhousel/maki/blob/mb-pages/src/basketball-24.svg

For c) and d) I was not able yet to produce a good looking 14px png export from any of the free svg sources I have found. Perhaps my fault, I am not an experienced icon designer. But I nevertheless wanted to submit this request already. Here are some horse icons:
- https://commons.wikimedia.org/wiki/File:Horseicon.svg
- https://github.com/openstreetmap/iD/blob/c8e4f76c230512dc82dc060bf337e3ba489b113e/svg/iD-sprite.src.svg used in iD for [bridleway](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dbridleway#Germany:_Public_ways_vs._forests.2Ffarmland) and [riding route](https://wiki.openstreetmap.org/wiki/Tag:route%3Dhorse)
- https://github.com/mapbox/maki/blob/master/icons/horse-riding-15.svg for [leisure=horse_riding](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dhorse_riding)
- https://de.wikipedia.org/wiki/Datei:Sinnbild_-_Reiter,_StVO_1970.svg
- https://commons.wikimedia.org/wiki/File:Sinnbild_Reiter.svg
- https://commons.wikimedia.org/wiki/File:Ratsastuspaikka_774f_tunnusosa.svg
 
Also please take into account that pitches can be indoors (in a hall or under a roof), in these cases markings can not be used, but icons probably can.

**Comment by imagico** [comment]
We would need to be careful here because it might not be clear what feature the icon applies to, i.e. the pitch itself or the facility the pitch is part of.

Also note the horse as a symbol - no matter if with a rider or not - can mean a lot of different things and does not necessarily imply something sport related.

**Comment by Hufkratzer** [comment]
Not every [riding arena](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch) (equestrian pitch) is part of a [riding facility](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dhorse_riding) or anything similar, like also not every other kind of pitch is part of a sports centre, leisure park or so. Especially in these cases it would be helpful if the map would show what kind of pitch it is.

I agree that it would be good if the icon would look related to equestrian sports, perhaps a jumping horse, but I don't know if that can be done in a 14px icon.

**Comment by boothym** [comment]
@kocio-pl I know I've asked before but any chance the icons I posted above could be rendered? 

I noticed there is a 0.3 opacity white outline to the black icon in the svg which I had to remove first. 
Preview I created in photoshop:
![tennis](https://user-images.githubusercontent.com/16965772/32737246-7b18b582-c891-11e7-9e17-7d9f2586034e.png)

**Comment by kocio-pl** [comment]
Sorry, I'm currently busy with many other problems, which I think are more important. 

There's not clear what people would decide, but the best chances would be to create 14 px versions of icons (leisure green is probably better than black - and we already have golf icon), test them, create the code and make a PR with renderings. Could you try to do it?

**Comment by Tomasz-W** [comment]
I was looking at Taginfo sport=* key list to choose which tags can have an special icon in combination with `leisure=sport_centre` / `leisure=pitch` . It's very interesting, beacuse there are some sports which I even didn't hear about earlier (eg. netball or gaelic games), more popular than eg. boxing or car racing. I think we should prepare a list of `sport=*` tags which we want special icon for, and for the rest use just generic leisure-green dot for `leisure=sports_centre` (see: https://github.com/gravitystorm/openstreetmap-carto/issues/2930) and current "blank" `leisure=pitch` rendering.
Which sport centres and pitches you would like to see with a special icon?

**Comment by Tomasz-W** [comment]
@HolgerJeromin We need sport icons for pitches anyway:
* there are 32k of `leisure=pitch` nodes 
*  there are also examples of linear ways mapped between two soccer goals, because there is no psychical "pitch", but just these two goals placed somewhere on a bigger grass area.

Examples:
https://www.openstreetmap.org/way/488863988
https://www.openstreetmap.org/way/454349719

![1](http://streamtube.pl/wp-content/uploads/2018/04/boisko-osiedlowe-os-batorego-now.jpg)
Unfortunely, most of mappers draw these "wild pitches" as `leisure=pitch` areas, which is wrong because theirs outlines are just imaginary and false.

Line markings propably won't resolve it (but of course we should try them on test renderings, besides of icons).

**Comment by HolgerJeromin** [comment]
> there are also examples of linear ways mapped between two soccer goals,

This is not documented
https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch

**Comment by Tomasz-W** [comment]
As tagging pitches as nodes and ways was always allowed by infobox, but not described, I've added some tagging instructions and examples:
https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch

So now it's documented.

**Comment by Adamant36** [comment]
@Tomasz-W any chance you could come up with usable icons for basketball and tennis? It would help a lot with mistagging if they could have icons.

**Comment by Tomasz-W** [comment]
@Adamant36 rescaled Maki icons:
* ![sport baseball](https://user-images.githubusercontent.com/25656654/43196564-a6c6c360-9008-11e8-8feb-03122dbf9f86.png) (baseball) -> Gist link: https://gist.github.com/Tomasz-W/ddcd4bb1b24650206414c97862cf0a1c
* ![sport basketball](https://user-images.githubusercontent.com/25656654/43196591-c2263ab4-9008-11e8-8b7f-0dec6ef3472b.png) (basketball) -> Gist link: https://gist.github.com/Tomasz-W/ff099f115b9b164a2f911d90c6d46be4
* ![sport soccer](https://user-images.githubusercontent.com/25656654/43196616-da7b95c8-9008-11e8-87e6-18ae7538800b.png) (soccer) -> Gist link: https://gist.github.com/Tomasz-W/0a3692589f27fb524977c47076a54e40
* ![sport tennis](https://user-images.githubusercontent.com/25656654/43196629-e6a1da06-9008-11e8-966f-89db4eed5dae.png) (tennis) -> Gist link: https://gist.github.com/Tomasz-W/56ca838fbcc41cf22843e51fa6d06532

I'll upload my alternative icons proposals later.

**Comment by matthijsmelissen** [comment]
Before we go into detail, do we think it would be desirable to add icons for specific sports?

**Comment by kocio-pl** [comment]
I agree. This is old question, but still relevant.

**Comment by wmyrda** [comment]
As former pro player I may be biassed here, but IMHO On z19 or z18, maybe even z17 they should not hurt anyone.

Related to that if someone is looking closely than on z13 and definitely with no issue on z15 could easily spot stadiums https://www.openstreetmap.org/#map=15/50.0288/21.9980 

As I live in the city I know where the sport gym for ~5000 spectators is, but could anyone else know looking at the above link? My guess is that sport icons could help with it.

**Comment by HolgerJeromin** [comment]
> As former pro player I may be biassed here, but IMHO On z19 or z18, maybe even z17 they should not hurt anyone.

Line markings hurts even less. Here 17:
![image](https://user-images.githubusercontent.com/2410353/43210425-62049ec4-902f-11e8-8f55-a19397ca44d4.png)

**Comment by Tomasz-W** [comment]
I'll repeat my question: What other sports you would like to see a special icon with for pitches?

**Comment by meased** [comment]
`american_football` is very popular in the US.

I like the idea of line markings, but I think it breaks down with baseball diamonds. It would be very difficult to get the orientation right and will probably end up with some strange renderings.

FYI, Apple Maps is evidently trying to do this:
https://techcrunch.com/2018/06/29/apple-is-rebuilding-maps-from-the-ground-up/
![apple-maps-sports-building-parking](https://user-images.githubusercontent.com/14190496/43221354-804bc9c6-9001-11e8-99a2-769d44f1ac50.png)

**Comment by Adamant36** [comment]
I've been dealing with names on the map going against the naming rule a lot lately. Especially as it relates to sports pitches and miss tagging like name=tennis court, name=basketball court, etc. I think adding icons so people can spot tell what kind of court they are on the map by the icon would solve a lot of that.

**Comment by meased** [comment]
The [french style](http://tile.openstreetmap.fr/) did *not* solve this issue, they just omitted baseball line markings:

![pitches](https://user-images.githubusercontent.com/14190496/43236739-9664e4c4-903b-11e8-84d3-c7959feee51f.png)

The football and soccer pitches have line markings, but baseball does not. (They also screwed up the orientation for football and soccer and frankly, it looks terrible...)

**Comment by Adamant36** [comment]
@meased My bad. It worked for the few pitches I tried it on. So I thought they had. I agree that it looks terrible though. In light of that, I retract my earlier statement about it ;)

**Comment by wmyrda** [comment]
@Tomasz-W If icons for sports mentioned in [post above](https://github.com/gravitystorm/openstreetmap-carto/issues/844#issuecomment-276772643) will be added than I will be satisfied. Most important from them for me would be the one for **multi** so it can be added later on for gyms. Gyms in most cases can be adjusted for many sports like basketball, volleyball, handball, futsal (soccer in gym), badminton to name just a few.

**Comment by HolgerJeromin** [comment]
The [german style](https://www.openstreetmap.de/karte.html?zoom=17&lat=45.72418&lon=-122.68659&layers=B000TT) has some fixes for this problem: 

![image](https://user-images.githubusercontent.com/2410353/43249524-3ed43c20-90bb-11e8-8748-c18277ba3720.png)

**Comment by wmyrda** [comment]
I actually considered lines as quite good idea and fine alternative to the icons, but when I see football field and tennis court I made up my mind. I would  much rather see the icon in this cases as those areas look like electric coil of some kind...

**Comment by Tomasz-W** [comment]
Please move discussion about pitch markings there: https://github.com/gravitystorm/openstreetmap-carto/issues/1126 . Let's discuss just the icons here.

**Comment by Hufkratzer** [comment]
As I wrote in https://github.com/gravitystorm/openstreetmap-carto/issues/2344, the proposed JOSM icon for sport=equestrian needs refinement. The current version doesn't look sporty at all, but rather looks like a beginner just before he falls off his horse.

**Comment by matthijsmelissen** [comment]
Once more: before starting to think about concrete icons, we should first decide whether we consider it desirable to add an icon for every single sport.

**Comment by Hufkratzer** [comment]
I consider it desirable because, as you explained at the beginning:
> Currently all sport fields are shown in the same style. It is very difficult to distinguish the different sports.

We have more than 60 different icons for different kinds of shops. So why should 20 sports icons be a problem?

**Comment by Tomasz-W** [comment]
On the one hand we should render some similar features the same to avoid map cluttering, but on the other hand we should render different object differently to show theirs functions properly.

In this case I'm sure we should add these icons for pitches, because function of eg. soccer pitch is totally different than skateboard one, and now they are both rendered the same. I'm just considering do we need special icons for different sports centre types, because very often they are mentioned in a club name, eg. _"Repton Boxing Club"_, so dot-rendering may be enough there.

**Comment by Hufkratzer** [comment]
First I thought that there could be a problem with too many icons if many table_tennis tables are near together like here:
![table_tennis](https://user-images.githubusercontent.com/5958184/43383856-970da1f0-93dc-11e8-9ba6-844d46077a0f.jpg)
(https://www.openstreetmap.org/way/302613825)

But then I saw that for other objects like picnic_tables and benches this problem seems to have been solved, see here:
![picnic_tables](https://user-images.githubusercontent.com/5958184/43384030-088ffdd2-93dd-11e8-8d79-7304d28493db.jpg)
(http://www.openstreetmap.org/node/1894660341 in iD;
the small icons are from iD, the bigger ones (less) from carto).

I wonder why this kind of solution wasn't/isn't applicable for swimming_pool's (https://github.com/gravitystorm/openstreetmap-carto/issues/2870).

**Comment by Tomasz-W** [comment]
In case of somebody will be interested in making a test renderings:
* Maki icons: https://gist.github.com/Tomasz-W/ddcd4bb1b24650206414c97862cf0a1c
* @Tomasz-W icons: https://gist.github.com/Tomasz-W/a56711ac75baf0bddf8655a04bdf8924

**Comment by Hufkratzer** [comment]
What about **sport=horse_racing** on **leisure=track**?

Wiki: 
* https://wiki.openstreetmap.org/wiki/Tag:sport=horse_racing
* https://wiki.openstreetmap.org/wiki/Riding

sport=horse_racing has > 2k uses just on tracks which is more than scuba_diving, canoe / rowing and billiards.each have in total and these tracks for horse_racing are quite large objects.

Examples: 
* [Hippodrome de Nancy-Brabois](https://www.openstreetmap.org/way/121289983
) contains [track1](https://www.openstreetmap.org/relation/1659972) and [track2](https://www.openstreetmap.org/relation/1659973)
* [Epsom Downs](https://www.openstreetmap.org/way/158859049)

Icon could be something like
![racingb14](https://user-images.githubusercontent.com/5958184/46873574-34410680-ce37-11e8-88dd-85e54ed36228.png) (just a rough draft) 
(:horse_racing: <- github horse_racing icon)

**Comment by HolgerJeromin** [comment]
> (🏇 <- github horse_racing icon)

No, this is the horse racing icon of your browser/ operating system.
Win 10 is quite colorful for example:
![image](https://user-images.githubusercontent.com/2410353/46885127-58144480-ce57-11e8-9a3e-dd0fbe87fa20.png)

**Comment by Adamant36** [comment]
@Hufkratzer, I think we could render a horse racing icon sports centers that have the sport=horse_racing tag. I don't don't recommend it on on tracks though due to the current issues we have with icons positions not rendering well if the area is round etc. Otherwise, it will render it will render in the middle or some other weird spot. If I remember correctly it was an issue on the Maptnik side that was supposedly fixed, but I've still been seeing it around sometimes. Its been a while though.

**Comment by Hufkratzer** [comment]
I think most tracks (running, athletics, horse_racing, skateboarding, also motorsport tracks (highway=raceway's)) are currently **not** in sports_centre's, examples https://www.openstreetmap.org/way/192989911
https://www.openstreetmap.org/relation/6154664
If you render icons for leisure=**sports_centre** instead of leisure=**track** you will somehow force mappers to map sports_centre's where no sports_centre's are (or to nest sports_centre's) if they want the sport specific icons on the map. Is that desirable?

Also leisure=**stadium** has no icon; if you render sports_centre with an icon and stadium **not**, you may encourage mappers to retag stadium's with leisure=sports_centre; is that desirable? Examples:
https://www.openstreetmap.org/way/258815225
https://www.openstreetmap.org/relation/8129847
https://www.openstreetmap.org/way/146537002. 

I can imagine that the placement of an icon may often be more complicated for tracks than for pitches or for sports_centre's, but If it's just a mapnik bug that hinders you to render icons for tracks, it may be better to wait until this bug is fixed, rather than to do it for sports_centre's instead. But when I look at this sports_centre it seems that the placement of icons works quite well:
https://www.openstreetmap.org/relation/8695416

**Comment by Adamant36** [comment]
Woho, I finally got the code for sports icons working. So expect some test renderings any day now. This ones been on the top of my list for a while now. 
![baseball](https://user-images.githubusercontent.com/30259065/48416389-ebfe5800-e704-11e8-87e4-e08e553a62ca.png)

**Comment by Adamant36** [comment]
Maki versus Tomasz-W icons. If people don't mind I think I'll do a PR for the main sports just to get to it implement and do the rest later. Since there's like 16 of them and it will take a lot more time otherwise. 

Maki baseball
![baseball maki](https://user-images.githubusercontent.com/30259065/48751578-f4d9b700-ec39-11e8-846c-67e062b65c4a.png)
Tomasz-W baseball
![baseball tomasz](https://user-images.githubusercontent.com/30259065/48751589-fdca8880-ec39-11e8-8376-590ef4f3632e.png)
Maki basketball
![basketball maki](https://user-images.githubusercontent.com/30259065/48751597-0e7afe80-ec3a-11e8-92c9-09ae21c1645c.png)
Tomasz-W basketball
![basketball tomasz](https://user-images.githubusercontent.com/30259065/48751618-366a6200-ec3a-11e8-8b89-e55ebe8cb2ea.png)
Maki soccer
![soccer maki](https://user-images.githubusercontent.com/30259065/48751623-3a967f80-ec3a-11e8-82c4-19e66b863159.png)
Tomasz-W soccer
![soccer tomasz](https://user-images.githubusercontent.com/30259065/48751627-471ad800-ec3a-11e8-8f89-3bc781926bc2.png)
Maki tennis
![tennis maki](https://user-images.githubusercontent.com/30259065/48751638-51d56d00-ec3a-11e8-9c6b-63ff8e02dca2.png)
Tomasz-W tennis
![tennis tomasz](https://user-images.githubusercontent.com/30259065/48751649-5e59c580-ec3a-11e8-8b84-e062f4dce499.png)

**Comment by Tomasz-W** [comment]
@Adamant36 For combination with `leisure=pitch` icons should be _man-made-grey_ (_leisure-green_ in combination with `leisure=sports_centre`)

**Comment by Tomasz-W** [comment]
@Adamant36 Imagine a big soccer training centre - if we would render both pitches and sports centres in leisure green, there will be just a chaotic bunch of green balls around, in my vision there would be man-made-grey pitches and green ball icon in sports centre middle. It was actually proposed initially: https://github.com/gravitystorm/openstreetmap-carto/issues/844#issuecomment-343980174. I know this ticket is about piches but we have to think long-term.

**Comment by meased** [comment]
I think https://github.com/gravitystorm/openstreetmap-carto/issues/844#issuecomment-343980174 was just a lazy photoshop example, not a request to use black icons. I find grey icons quite odd as I can't see how something like "baseball" would be categorized as "generic" and not "leisure".

Could you explain your vision a little better? Perhaps a mockup?

**Comment by geostonemarten** [comment]
And what about multi or comma separated value. In your case you don't add name but I have lot of example with complexe name and sub   parts with pitch name

**Comment by kocio-pl** [comment]
I would try with sport icons for pitches rendered as green, but stadiums or other bigger sport entities with a reversed white icon on a green background -  something like we do with bus stops and bus stations:

![bus station](https://wiki.openstreetmap.org/w/images/thumb/5/5a/Amenity_bus_station.svg/120px-Amenity_bus_station.svg.png)

**Comment by geostonemarten** [comment]
> I would try with sport icons for pitches rendered as green, but stadiums or other bigger sport entities with a reversed white icon on a green background - something like we do with bus stops and bus stations:
> 
> ![bus station](https://camo.githubusercontent.com/7bed10c7aaba5723394960ffc312cabae8c86ccd/68747470733a2f2f77696b692e6f70656e7374726565746d61702e6f72672f772f696d616765732f7468756d622f352f35612f416d656e6974795f6275735f73746174696f6e2e7376672f31323070782d416d656e6974795f6275735f73746174696f6e2e7376672e706e67)

Why??? that is not the case for all other icons in openstreetmap carto ...

**Comment by kocio-pl** [comment]
Because this way we would keep the color for pitches while at the same time indicating that they are special (bigger).

**Comment by boothym** [comment]
@Adamant36 agreed, though my only concern with the Tomasz-W icons is the basketball being too similar to the soccer ball?

Do you mind rendering a pitch which has a name as well as a sport?

@Tomasz-W my image further up the thread was just a quick photoshop to try and get some progress on this issue (first comment in 9 months!) 😉 I'd go for green now that I've seen it rendered.

As for a big soccer training centre - it would be good to have some example renderings of both pitches and sports centres with sport=*:

- https://www.openstreetmap.org/way/145858940
- https://www.openstreetmap.org/#map=15/53.4238/-2.3678
- https://www.openstreetmap.org/way/428742471 (weird that the name doesn't render)

**Comment by Adamant36** [comment]
So, green for the icons it is then.........Any feedback besides @boothym's on which set of icons are better? 

@boothym, they do kind of look similar. I'm willing to go with Maki's soccer ball, if there is wider agreement. 

Also, as I said above, I'm not doing sports centers until pitches get done and rendered. its a lot less complicated that way. Plus, we haven't even figured out the icons yet anyway. So, its kind of putting the cart before the horse. If there's a problem with that, someone else is free to work on the issue.

**Comment by Tomasz-W** [comment]
My picks of icons after test renderings:

* baseball -> Tomasz-W
* basketball -> 50/50
* soccer -> Tomasz-W
* tennis -> Tomasz-W

@Adamant36 Anyway, I hope you will make tests of example places linked above in both versions to compare and make a fair decision ;)

**Comment by Adamant36** [comment]
@tomasz-W, OK. 

I will eventually after pitches get done. Rendering on sports centers doesn't really affect how or what's rendered on pitches at this point though. Plus, The green dot-rendering is a different issue. It will get done at some point in its own PR.

**Comment by Adamant36** [comment]
@Tomasz-W, btw, do you know of any sports centers tagged as one sport that contains pitches from another sport? Also, it seems like it would be a little redundant to have an icon for a sports center along with all the pitches there. It should be clear from the pitches what kind of sports center it is. Plus, they might block each other out. 

P.S. Can you make icons for roller skating and ice skating? They both have numbers around 1000 each, but I feel like they could be added anyway. Since there probably aren't many places for them in the world anyway.

**Comment by Hufkratzer** [comment]
> my only concern with the Tomasz-W icons is the basketball being too similar to the soccer ball?

The maki basketball icon is hard to recongnize, therefore I would prefer the ball icon for basketball.

What about something like ![mybb14](https://user-images.githubusercontent.com/5958184/49253192-506c2780-f426-11e8-8e76-29f522a02364.png)? This is from https://commons.wikimedia.org/wiki/File:Font_Awesome_5_solid_basketball-ball.svg,
not free, would have to be redrawn.

**Comment by IgorEliezer** [comment]
Hello, :wave: 

If having so many icons is an issue (#3651) or if having field marking is too much (#1126), why don't you people reuse the `leisure=fitness_station` icon to represent all ground sports since most of them involve running or at least moving on a field?

I think having to distinguish a sport field from the rest of the map is more of an issue than distinguishing the fields from each other.

It would at least reduce the need of using generic names like "Soccer field" to identify a green rectangle in the middle of nowhere, like this https://www.openstreetmap.org/#map=17/-23.42452/-47.58841.

**Comment by Adamant36** [comment]
@imagico, what would you think about @IgorEliezer suggestion? I was also thinking maybe it would doable to at least add icons for sports complexes. Since it wouldn't lead to that much more clutter, there's a few different kind of sports that have sports complexes in the first place, and there's already an icon for swimming centers. What's your opinion doing that? At least then we could not go off the cliff of rendering icons for everything from baseball to mud wrestling.

**Comment by imagico** [comment]
This issue is about differentiating sport pitches by type of sport.  I don't think the suggestion would do that.

The running figure symbol - which is currently used for fitness centres and sports shops - is one of the more problematic symbols currently used by this style.  The running figure is internationally most commonly used for symbolizing emergency exits - see for example:

https://commons.wikimedia.org/wiki/File:Ketheltunnel_nooduitgang.jpg
https://commons.wikimedia.org/wiki/File:Slovenia_road_sign_III-122.1.svg

Using this to symbolize sports has a high probability of being misleading and confusion - already in how it is used right now but further if this use was extended.

**Comment by kocio-pl** [comment]
I believe basket icon has the same probability of confusion, because from what I remember it's commonly - if not exclusively - used as "shop" icon (not "convenience shop"). The context (being inside buildings or on the fresh air, not on the edges of buildings, plus we have totally different symbols for entries) makes running figure safe for me.

I don't have a strong opinion how to render sport grounds, both having individual icons (at high zoom level) or reusing running man icon is OK for me, I just prefer the first one.

**Comment by pqhf5kd** [comment]
Any updates on this?

**Comment by imagico** [comment]
The current problem with this just like with any other POI symbol additions is the lack of consensus on #3635.  The best chance if anyone wants to address this would probably be to work towards a symbolization other than a POI icon.  There was fairly elaborate discussion also on #3651 that might be helpful in that regard.

**Comment by Adamant36** [comment]
> Any updates on this?

I tried it in #3651 and it was way to busy. As @imagico says, the best way forward is probably without it involving a POI icon. Which kind of makes me think this issue should be closed as wontfix.

**Comment by imagico** [comment]
I changed the issue title to not imply a POI icon is necessarily the only solution.  IMO differentiating sports would still be a useful feature if it can be implemented in a non-obtrusive way.

**Comment by Adamant36** [comment]
What about just rendering the fill color of pitches based on the surface or what the sport tag is? I feel like pitch green is to all encompassing and has homogenized a bunch of things that barely share anything in common in real life. If all the sports where not connected through them having the pitch tag they would probably be rendered completely differently. There's more then enough uses for a lot of them to consider them as separate objects. I think pitch green works when its the main sports that take place on grass like baseball, football, and soccer. When it includes things like beach volleyball, horse ridding, or skate parks, none of which are usually done on grass or considered pitches, it's kind of to generalized and loses its thematic meaning.

**Comment by jeisenbe** [comment]
You are correct that some uses of leisure=pitch are not usually
covered in grass. However, there are some sports where more than one
surface is possible: for example, tennis courts are often grass, but
can also be artificial turf, concrete or clay. And many pitches which
are usually grass can be replaced by artificial turf, or might be just
dirt in less developed countries.

I would consider the sport to be the more important information: if
you want to play basketball you will be disappointed by a tennis or
volleyball court, whether it is grass, clay or concrete.

**Comment by Adamant36** [comment]
> I would consider the sport to be the more important information: if
you want to play basketball you will be disappointed by a tennis or
volleyball court, whether it is grass, clay or concrete.

At least with the main sports, I think most people can extrapolate what it is by the size, shape, and surrounding details. No one is going to confuse a baseball field for a basketball court, an American football field for a tennis court., an American football field for a baseball field or whatever. Having the leisure pitch green color on them just seems to add a pointless extra bit of cognitive load. Which isn't really that intuitive anymore with over 200 sports being tagged with it, that share nothing in common with each other, except for all being called "sports." But "sport" is mainly a mental category, not a cartographic (geographic) one. Whereas, the rendering is currently as if it's a cohesive, shared by all sports, landuse (called "pitches") or something similar. Which it isn't IMO.

**Comment by jidanni** [comment]
See also https://wiki.openstreetmap.org/wiki/Talk:Tag:leisure%3Dpitch#Rendering_of_pitches_according_to_sport .

**Comment by HolgerJeromin** [comment]
@jidanni 
This alternate solution in tracked in #1126

**Comment by imagico** [comment]
I have shown now a demonstration for a possible approach to this in

http://blog.imagico.de/single-symbol-patterns-in-maps/

I am not going to submit this as a PR here as is because i am not sure if this would be a good change under the goals we have and there are numerous issues that would need to be looked at.  But it is a demonstration technically and design wise how you can approach this issue without resorting to the primitive POI symbol solution so far discussed which as is being discussed in #3635 is non-sustainable to extend further in this style.  Someone interested in differentiating sport pitches might take this as inspiration to develop a solution here.  In any case if you do so please consider the points mentioned in https://github.com/gravitystorm/openstreetmap-carto/pull/3651#issuecomment-455859844.

**Comment by Adamant36** [comment]
Genius solution. How do you think it would conflict with the goals of the style and what issues need to be looked at before it can be implemented?

**Comment by jeisenbe** [comment]
Symbol examples:

![sport_symbols_list_s](https://user-images.githubusercontent.com/42757252/85358751-64282480-b4c9-11ea-818b-c1a1a3a005a1.png)

Looks quite good. I would be happy to see a PR which implements this idea.

**Comment by jeisenbe** [comment]
Rendering sample from the [link above](http://blog.imagico.de/single-symbol-patterns-in-maps/):

![sport_symbols_div_z18](https://user-images.githubusercontent.com/42757252/85358856-96398680-b4c9-11ea-830a-11ca50b7b895.png)

**Comment by jeisenbe** [comment]
This sort of symbology could also be a good solution for dog parks, which currently have a repeated symbol which leads to an excessively strong pattern.

**Comment by imagico** [comment]
The issue of `sport=*` being tagged on nodes and on features other than `leisure=pitch` is not addressed by this - see https://github.com/gravitystorm/openstreetmap-carto/pull/3651#issuecomment-455859844.

**Comment by kocio-pl** [comment]
I like the way it looks for pitch areas - it enhances mapper feedback and at the same time is visually unobtrusive (unlike straight icons proposed earlier).

As of nodes - since we're currently not rendering them, we don't have to start to render them from now on.

**Comment by kocio-pl** [comment]
How this rendering would work with named objects ([~19% of `sport=*` tags have name](https://taginfo.openstreetmap.org/keys/sport#combinations))?

**Comment by imagico** [comment]
The change discussed does not modify label rendering for pitches - see https://github.com/gravitystorm/openstreetmap-carto/commit/fc2657db1b5325a669626081819857941d1ffea8.

Regarding labeling of pitches - see #4210.

**Comment by kocio-pl** [comment]
Is anybody willing to prepare such PR? I'd like to test it.

**Comment by Hufkratzer** [comment]
As alredy mentiones earlier (in https://github.com/gravitystorm/openstreetmap-carto/issues/844#issuecomment-276772643):

Please don't forget a symbol for  [sport=equestrian](https://wiki.openstreetmap.org/wiki/Tag:sport%3Dequestrian).

The current taginfo values for sport in combination with pitch are:
| value | frequency |
| ------------- | ------------- |
| soccer | 443k  |
| tennis | 356k  |
| basketball | 159k  
| baseball | 134k | 
| multi | 67k | 
| **equestrian** | 32k | 
| beachvolleyball | 21k | 
| volleyball | 20k | 
| skateboard | 16k | 
| american_football | 16k | 
| boules | 15k | 
| table_tennis | 13k |  
| bowls | 12k | 
| athletics | 9k | 
| cricket | 9k | 
| shooting | 5k | 

A possible symbol for "equestrian" may be found in https://github.com/gravitystorm/openstreetmap-carto/issues/2344#issuecomment-427694146

**Comment by sekerob** [comment]
Hi, apologies for waking up this thread again, really liking your rendering design and often revisit to see how things come out, noticed that sometimes a pitch of a certain sport does not render, just the green square and wondered if there is some minimum size. Have a situation where a basketball pitch renders at a school, but not at a park 2 miles away where a same size tennis court does render. Another where a full size soccer pitch renders, but not the 'street soccer' sized at same school and park. Sample http://tile.openstreetmap.fr/?zoom=19&lat=42.46991&lon=14.20103&layers=B0000000FFFFFFF

thanks for reading.

**Comment by imagico** [comment]
Your link goes to the french style (https://github.com/cquest/osmfr-cartocss).  You will have to inquire about issues with their pitch rendering there.

**Comment by michjans** [comment]
I was referred to this issue from the Open Street Map community forum, because I asked the question why I didn't see a skating icon on a skating pitch that I've just added.

I used the following tags:
leisure=pitch
sport=skateboard

Just wanted to bring this under attention again. Apparently it is more complicated to solve than it seems?

**Comment by dch0ph** [comment]
> Apparently it is more complicated to solve than it seems?

Not necessarily complicated, but not necessarily straightforward either. A previous PR (#3651) didn't achieve consensus support and nobody else has since taken on the challenge.

@imagico demonstrated a [proof of concept](http://blog.imagico.de/single-symbol-patterns-in-maps/), which was well received, but he felt that there were [too many open issues to submit a PR](https://github.com/gravitystorm/openstreetmap-carto/issues/844#issuecomment-647704856). As I read it, some of these open issues were (quoting):

- _"Rendering nodes with symbol only but without a generic symbol for leisure=pitch alone is counter-intuitive for the mapper and incentivizes against one feature, one OSM element (by suggesting to tag multi-use pitches with several pitch nodes)"_
`sport=multi` (67k uses above) would probably need a distinctive icon to avoid this. 

- _"Selectively rendering leisure=pitch + sport=* might also be counterproductive for mapping indoor sports infrastructure - for which tagging sport on leisure=pitch probably is not the most common method (there are for example 82k combinations leisure=sports_centre + sport'=*). That would deserve more thorough analysis."_ Personally, I'm not too worried about this - I think differentiating pitches is interesting / important in its own right. I don't see that it would harm / affect indoor sport mapping.

- _"The lack of a sustainable overall concept for rendering sport pitches and point symbols in general indicating an inflation similar to as it has happened for shops."_ I feel this goes with the territory - if we think that adding sport icons would improve the map, then there are going to be more icons. Shops are always going to be a bigger problem.

One that I can think of:
* We would also need to decide whether to have a specific icon for "pitch without sport" (as we do for wood without leaf type) or, alternatively, an icon for "unrecognised sport" (i.e. sport specified, but without a symbol to render).
 
Note that the Tracestack Topo rendering does show sport icons. The approach used seems to follow #3651?

**Comment by xenjogmbh** [comment]
I run a yoga studio (Yoga Albisrieden, Zürich) tagged as leisure=fitness_centre + sport=yoga. Currently it renders with a generic runner icon, which is misleading for users looking for yoga-specific facilities. A dedicated yoga symbol (seated figure / lotus pose) would greatly improve discoverability for studios like mine. This is a real-world use case affecting many yoga studios across OSM.

**Comment by imagico** [comment]
`leisure=fitness_centre` was not on my radar for this, but it is definitely covered by this issue.

`leisure=fitness_centre` has about 15 percent of features with `sport=*` specified, 7846 with `sport=fitness` and more than 6000 with `sport=yoga` (which is the dominant use case for `sport=yoga` - which has a total of 7000 uses).

The challenge is going to be for symbology to distinguish between `leisure=sport_centre` and `leisure=fitness_centre` in an intuitive way.  The difference between the two is fairly well established practically but not really documented.  For `sport=yoga` this is not much of a problem since that is not widely tagged on `leisure=sport_centre` - but for other `sport=*` this is an issue.  The current symbol for `leisure=fitness_centre` is definitely not a very good choice in that regard since there are plenty of `leisure=sport_centre` mapped that facilitate running sports.

**Comment by Hufkratzer** [comment]
> A dedicated yoga symbol (seated figure / lotus pose) would greatly improve discoverability for studios

I wouldn't associate someone sitting around in lotus pose with physical fitness or sport or yoga as a sport, rather an icon like this:

<img width="96" height="93" alt="Image" src="https://github.com/user-attachments/assets/ec58bd61-59d6-4e0a-aec3-2030dfaeb809" />

from https://thenounproject.com/icon/yoga-6648644/

> The challenge is going to be for symbology to distinguish between leisure=sport_centre and leisure=fitness_centre in an intuitive way. The difference between the two is fairly well established practically

iD uses another icon for fitness_centre without sport=*:

<img width="628" height="155" alt="Image" src="https://github.com/user-attachments/assets/44806802-e68a-4300-acc0-892fa21c12a8" />

but it changes to a runner icon if sport=yoga is added.

Tracestrack Topo uses

<img width="440" height="99" alt="Image" src="https://github.com/user-attachments/assets/05e70ed1-ffbc-46da-b17a-2d3ce741ea01" />

for fitness_centre + sport=yoga

See also old [discussion](https://github.com/openstreetmap-carto/openstreetmap-carto/issues/2375#issuecomment-257988155)

**Comment by pqhf5kd** [comment]
This looks like a ballet dancer to me

**Comment by dch0ph** [comment]
> I wouldn't associate someone sitting around in lotus pose with physical fitness or sport or yoga as a sport,

But `leisure=fitness_centre` + `sport=yoga` is how mappers are choosing to map yoga studios, and the question is the most appropriate symbol.

The advantage of a sitting person is exactly that it is unlikely to be confused with other sport-related activities, such as ballet dancing (or figure skating?)!
